be2-scala/src/main/scala/ch/epfl/pop/model/objects/PublicKey.scala

```diff
@@ -1,9 +1,34 @@
 package ch.epfl.pop.model.objects
 
+import ch.epfl.dedis.lib.crypto.{Ed25519, Ed25519Point, Ed25519Scalar}
+
+import java.security.SecureRandom
+import PublicKey._
+
 import java.security.MessageDigest
 
 final case class PublicKey(base64Data: Base64Data) {
+
+  def encrypt(messageB64: Base64Data): Base64Data = {
+    val asPoint = new Ed25519Point(base64Data.decode())
+    val message = messageB64.decode()
+    if (message.length > MAX_MESSAGE_LENGTH) throw new IllegalArgumentException(s"The message should contain at maximum $MAX_MESSAGE_LENGTH bytes")
+    val M = Ed25519Point.embed(message)
```

Comment:
I would use more expressive variable names here, for an outsider/a programmer onboarding the project it can be quite confusing

Reply:
However, the separation of the operations into different lines is good